activerecord/lib/active_record.rb

```
@@ -197,6 +197,36 @@ def self.default_timezone=(default_timezone)
 
   self.default_timezone = :utc
 
+  # The action to take when database query produces warning.
+  # Must be one of :ignore, :log, :raise, :report, or a custom proc.
+  # The default is :ignore.
+  singleton_class.attr_reader :db_warnings_action
+
+  def self.db_warnings_action=(action)
+    @db_warnings_action = case action
+                          when :ignore
+                            ->(_) { }
+                          when :log
+                            ->(warning) do
+                              warning_message = "[#{warning.class}] #{warning.message}"
+                              warning_message += " (#{warning.code})" if warning.code
+                              ActiveRecord::Base.logger.warn(warning_message)
+                            end
+                          when :raise
+                            ->(warning) { raise warning }
+                          when :report
+                            ->(warning) { Rails.error.report(warning, handled: true) }
+                          when Proc
+                            action
+    end
```

+ `else`; we don't want to silently ignore typos / unexpected classes.